### PR TITLE
[ios] Add removeStyleImage to MGLMapView

### LINKED ISSUE #14731: [ios, macos] Eviction of cached images bindings for darwin
The https://github.com/mapbox/mapbox-gl-native/pull/14610 pr introduced a new core api for evicting cached images when the images are provided through the `onStyleImageMissing` callback. This ticket tracks the implementation on Darwin.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Fixes #14731

### CODE REVIEW

**Comment by julianrex** [comment]
We should wait for https://github.com/mapbox/mapbox-gl-native/pull/14763 to land, as it moves a few things around.

File: platform/ios/src/MGLMapViewDelegate.h [hunk 1]
```diff
@@ -277,6 +277,17 @@ NS_ASSUME_NONNULL_BEGIN
...
+/**
+ Asks the delegate whether the map view should evict cached images.
+ 
+ This method is called in two scenarios: when the cumulative size of unused images
+ goes over the cache size or when the image's requesting tile is destroyed.
+ 
+ @param mapView The map view that is evicting the image.
+ @param imageName The image name that is going to be removed.
+ */
+- (BOOL)mapView:(MGLMapView *)mapView removeStyleImage:(NSString *)imageName;
```
Comment (1ec5) [inline platform/ios/src/MGLMapViewDelegate.h]:
```suggestion
- (BOOL)mapView:(MGLMapView *)mapView shouldRemoveStyleImage:(NSString *)imageName;
```

File: platform/macos/src/MGLMapViewDelegate.h [hunk 2]
```diff
@@ -333,6 +333,17 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (nullable NSViewController *)mapView:(MGLMapView *)mapView calloutViewControllerForAnnotation:(id <MGLAnnotation>)annotation;
 
+/**
+ Asks the delegate whether the map view should evict cached images.
+ 
+ This method is called in two scenarios: when the cumulative size of unused images
+ goes over the cache size or when the image's requesting tile is destroyed.
+ 
+ @param mapView The map view that is evicting the image.
+ @param imageName The image name that is going to be removed.
```
Comment (1ec5) [inline platform/macos/src/MGLMapViewDelegate.h]: Nit: Document the return value as well. (×2)

File: platform/macos/src/MGLMapViewDelegate.h [hunk 3]
```diff
@@ -333,6 +333,17 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable NSViewController *)mapView:(MGLMapView *)mapView calloutViewControllerForAnnotation:(id <MGLAnnotation>)annotation;
 
+/**
+ Asks the delegate whether the map view should evict cached images.
+ 
+ This method is called in two scenarios: when the cumulative size of unused images
+ goes over the cache size or when the image's requesting tile is destroyed.
```
Comment (1ec5) [inline platform/macos/src/MGLMapViewDelegate.h]: > This method is called in two scenarios: when the cumulative size of unused images exceeds the cache size or when the last tile that includes the image is removed from memory.

(×2)

File: platform/macos/src/MGLMapViewDelegate.h [hunk 4]
```diff
@@ -333,6 +333,17 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable NSViewController *)mapView:(MGLMapView *)mapView calloutViewControllerForAnnotation:(id <MGLAnnotation>)annotation;
 
+/**
```
Comment (1ec5) [inline platform/macos/src/MGLMapViewDelegate.h]: This method appears under the “Managing Callout Popovers” heading in the header and in documentation:

https://github.com/mapbox/mapbox-gl-native/blob/12cb77e5f22a05d66ad7fe30fd0bf62d50948214/platform/macos/src/MGLMapViewDelegate.h#L293

Create a new heading for “Managing the Style” or move this method up to the “Loading the Map” section.

File: platform/ios/src/MGLMapViewDelegate.h [hunk 5]
```diff
@@ -277,6 +277,17 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (nullable UIImage *)mapView:(MGLMapView *)mapView didFailToLoadImage:(NSString *)imageName;
 
+/**
```
Comment (1ec5) [inline platform/ios/src/MGLMapViewDelegate.h]: This method appears under the “Managing Callout Views” heading in the header and in documentation:

https://github.com/mapbox/mapbox-gl-native/blob/12cb77e5f22a05d66ad7fe30fd0bf62d50948214/platform/ios/src/MGLMapViewDelegate.h#L598

Create a new heading for “Managing the Style” or move this method up to the “Loading the Map” section.

Comment (fabian-guerra) [reply platform/ios/src/MGLMapViewDelegate.h]: On iOS is under the right header.

**Pull request merged**